sdk/src/main/java/io/horizen/sc2sc/Sc2ScUtils.scala

```diff
@@ -141,6 +131,18 @@ trait Sc2ScUtils[
     }
   }
 
+  private def buildEntireMerklePath(networkParams: NetworkParams, mcBlockRef: MainchainBlockReference, topCert: WithdrawalEpochCertificate): Array[Byte] = {
+    val existenceProof = mcBlockRef.data.existenceProof.getOrElse(
+      throw new IllegalArgumentException(s"There is no existence proof in the MainchainBlockReference $mcBlockRef")
+    )
+    Using.resource(
+      mcBlockRef.data.commitmentTree(networkParams.sidechainId, networkParams.sidechainCreationVersion).commitmentTree
+    ) { commTree =>
+      val pathCert = commTree.getScCommitmentCertPath(networkParams.sidechainId, topCert.bytes).get()
+      pathCert.updateScCommitmentPath(MerklePath.deserialize(existenceProof))
+      pathCert.serialize()
```

Comment:
please check but I think is needed to free this resource after the serialze
pathCert.freeScCommitmentCertPath()

Reply:
yes, you're right, gonna change it